File: LearnAppServerAPI/LearnAppServerAPI/Data/Entities/Flashcard.cs
﻿namespace LearnAppServerAPI.Data.Entities
{
    public class Flashcard
    {
        public int Id { get; set; }
        public string Front { get; set; }
        public string Back { get; set; }
        public int FlashcardsSetId { get; set; }
    }
}

File: mobileApp/lib/data/flashcard.dart
// ignore_for_file: non_constant_identifier_names

class Flashcard {
  int Id = 10;
  String Front = '';
  String Back = '';
  int FlashcardsSetId = 0;

  Flashcard(this.Id, this.Front, this.Back, this.FlashcardsSetId);

  Flashcard.fromJson(Map<String, dynamic> flashcardMap) {
    Id = flashcardMap['id'] ?? 0;
    Front = flashcardMap['front'] ?? '';
    Back = flashcardMap['back'] ?? '';
    FlashcardsSetId = flashcardMap['flashcardsSetId'] ?? 0;
  }

  Map<String, dynamic> toJson() => {
        'front': Front,
        'back': Back,
        'flashcardsSetId': FlashcardsSetId,
      };
}
